src/ripple/app/misc/AmendmentTableImpl.cpp

```diff
@@ -26,36 +26,162 @@
 #include <ripple/protocol/JsonFields.h>
 #include <ripple/protocol/TxFlags.h>
 #include <boost/format.hpp>
-#include <boost/tokenizer.hpp>
+#include <boost/regex.hpp>
 #include <algorithm>
 #include <mutex>
 
 namespace ripple {
+
+/** Amendments that this server supports and enables by default */
+static
+Section
+preEnabledAmendments ()
+{
+    Section section ("Pre-Enabled Amendments");
+
+    return section;
+}
+
+/** Amendments that this server supports, but doesn't enable by default */
+static
+Section
+supportedAmendments ()
+{
+    Section section ("Supported Amendments");
+
+    return section;
+}
+
+static
+std::vector<std::pair<uint256, std::string>>
+parseSection (Section const& section)
+{
+    // <key> '=' <value>
+    static boost::regex const re1 (
+            "^"                    // start of line
+            "(?:\\s*)"             // whitespace (optional)
+            "([abcdefABCDEF0-9]*)" // <hex amend ID>
+            "(?:\\s*)"             // whitespace (optional)
+            "(?:.*\\S+)"           // <value>
+        , boost::regex_constants::optimize
```

Comment:
I don't think this regular expression works:

1) The `value` group is never captured, and it is referred to in the code below.
2) If we remove the `?:` and capture the value group, it captures too much, including the `=` delimiter between the key and value
3) It will match any delimiter will work, not just an `=`
4) I prefer `\s*` to `(?:\s*)`
5) Since we call `SetHexExact` below, we should constrain the key to the expected number of characters
5) Raw strings can get rid of some of the backslashes

How about something like: `R"(^\s*([a-fA-F0-9]{64})\s*=\s*(\S+))"`